### ISSUE TITLE
Cannot update bot avatar

### ISSUE THREAD

**Issue by Scorr** [RogueException/Discord.Net#443]
In a command:
```cs
var fileStream = new FileStream(Directory.GetCurrentDirectory() + "/newAvatar.jpg", FileMode.Open);
var image = new Image(fileStream);
await Context.Client.CurrentUser.ModifyAsync(u => u.Avatar = image);
```
Any operations after the last line will not run, just seems to hang. No errors or timeouts.
In my case I confirmed that filestream is valid and image returns a proper Image.

edit: running 1.0.0-beta2-00523

**Comment by khionu** [comment]
Is your RunMode set to Async?

**Comment by Scorr** [comment]
Yeah, seems to be same problem with Sync though.

**Comment by khionu** [comment]
Have you put ConsoleWrites to ensure that it's completing each of those lines?

**Comment by Scorr** [comment]
Actually when trying the local file with Sync I'm getting
`Error getting value from 'ReadTimeout' on 'System.IO.FileStream'.`
when performing the ModifyAsync()
Thought this might be an user error, but dunno what would cause it since no errors occur until ModifyAsync().
Grabbing a memorystream from a downloaded message (which is what I had been using for testing) doesn't return that error on async/sync, also just hangs. Same for local file with async (like original example)

It's completing all of the lines and creating a new Discord.Image (although that just seems to be storing a stream).

**Comment by khionu** [comment]
Can you get the full stacktrace on that?

**Comment by Scorr** [comment]
Github preview seemed to not like the wall of text so threw it on hastebin: https://hastebin.com/usoruqavet.cs

**Comment by Scorr** [comment]
Also of note is that `Exception thrown: 'Newtonsoft.Json.JsonSerializationException' in System.Private.CoreLib.ni.dll` also occurs during async and the download/memorystream way, it just doesn't seem to log to discord or actually break and give a stacktrace in that case.

**Comment by khionu** [comment]
Try using the override for the Image CTOR that accepts the FilePath?

**Comment by Scorr** [comment]
Tried that as well; identical error.

**Comment by khionu** [comment]
Hmm, well, I can't repro due to #445, but we'll look into it after that's resolved

**Comment by LassieME** [comment]
I could repro this just fine, seems like the `DiscordContractResolver` is picking the wrong converter; `OptionalConverter` instead of `ImageConverter`. Rewriting the ContractResolver is a bit beyond me tho.

**Comment by Scorr** [comment]
Fixed by 3ddc59c
